lib/checkleakautovar.cpp

```diff
@@ -600,16 +600,37 @@ void CheckLeakAutoVar::checkScope(const Token * const startToken,
                 endDeleterToken = typeEndTok->linkAt(2);
             }
             if (deleterToken) {
+                // Skip the decaying plus in expressions like +[](T*){}
+                if (deleterToken->str() == "+") {
+                    deleterToken = deleterToken->next();
+                }
                 // Check if its a pointer to a function
                 const Token * dtok = Token::findmatch(deleterToken, "& %name%", endDeleterToken);
                 if (dtok) {
                     af = _settings->library.dealloc(dtok->tokAt(1));
                 } else {
+                    const Token * tscopeStart = nullptr;
+                    const Token * tscopeEnd = nullptr;
+                    // If the deleter is a lambda, check if it calls the dealloc function
+                    if (deleterToken->str() == "[" && 
+                        deleterToken->link() && 
```

Comment:
I would use Token::simpleMatch in this condition:

    if (deleterToken->str() == "[" &&
       Token::simpleMatch(deleterToken->link(), "] (") &&
       Token::simpleMatch(deleterToken->link()->linkAt(1), ") {"))